runtime/lua/vim/lsp.lua

```diff
@@ -446,8 +450,9 @@ lsp.config = setmetatable({ _configs = {} }, {
       for _, v in ipairs(api.nvim_get_runtime_file(('lsp/%s.lua'):format(name), true)) do
         local config = assert(loadfile(v))() ---@type any?
         if type(config) == 'table' then
+          rtp_config = rtp_config or {}
           --- @type vim.lsp.Config?
-          rtp_config = vim.tbl_deep_extend('force', rtp_config or {}, config)
+          rtp_config = vim.tbl_deep_extend(rtp_config.merge_behavior or 'force', rtp_config, config)
```

Comment:
Not sure if we want to limit `merge_behavior` to appearing at most once for each configuration, or if we should only allow it to be configured with `vim.lsp.config('*', ...)`. WDYT @lewis6991 @justinmk?